compiler/z/codegen/BinaryAnalyser.cpp

```diff
@@ -227,7 +227,17 @@ TR_S390BinaryAnalyser::genericAnalyser(TR::Node * root,
                tempMR->enforce4KDisplacementLimit(secondChild, cg(), NULL);
                }
 
-            generateRXInstruction(cg(), memToRegOpCode, root, thirdReg, tempMR);
+            auto instructionFormat = TR::InstOpCode(memToRegOpCode).getInstructionFormat();
```

Comment:
The choice between RX and RXE instructions should really exist. Looking at the two use cases of this pattern in BinaryAnalyser.cpp and BinaryCommutativeAnalser.cpp the code is organized in such a way that forces us to distinguish between the two. Examining the code further it looks like it can definitely be simplified so such checks are not even necessary.

They are added here in this way to satisfy the asserts which are added to make sure we're not using the wrong `generate*Instruction` API. I would prefer not to over complicate these APIs as it would set a dangerous precedent (i.e. why not have a `generateRXorRXEorRXYorRXSInstruction`? - which one are we generating then?).

RX and RXE instructions are inherently different formats. One has a mask where the other one doesn't. As opposed to RX and RXY which have the exact same operands except the RXY form has a slightly larger displacement.

Reply:
I'm going to try and fix this up in the followup PRs where I will clean up the remanences of things we deprecate here (extended code base register, etc.).